notifications_utils/formatters.py

```diff
@@ -287,6 +288,83 @@ def normalise_whitespace(value):
     return ' '.join(strip_and_remove_obscure_whitespace(value).split())
 
 
+def get_action_links(html: str) -> list[str]:
+    """Get the action links from the html email body and return them as a list. (insert_action_link helper)"""
+    # set regex to find action link in html, should look like this:
+    # &gt;&gt;<a ...>link_text</a>
+    action_link_regex = re.compile(
+        r'(>|(&gt;)){2}(<a style=".+?" href=".+?"( title=".+?")? target="_blank">)(.*?</a>)'
+    )
+
+    return re.findall(action_link_regex, html)
```

Comment:
. . . and then [return an iterator of regex Match instances](https://docs.python.org/3.10/library/re.html#re.finditer) rather than a list of strings.